include/pybind11/common.h

```diff
@@ -536,9 +536,13 @@ using is_template_base_of = decltype(is_template_base_of_impl<Base>::check((remo
 struct is_template_base_of : decltype(is_template_base_of_impl<Base>::check((remove_cv_t<T>*)nullptr)) { };
 #endif
 
+/// Check if T is an instantiation of the template `Class`. For example:
+/// `is_<std::shared_ptr, T>` is true if `T == std::shared_ptr<U>` where U can be anything.
+template <template<typename...> class Class, typename T> struct is_ : std::false_type { };
+template <template<typename...> class C, typename... Us> struct is_<C, C<Us...>> : std::true_type { };
```

Comment:
True, renamed. I kind of liked the naming as a prefix of some kind, e.g. `is_call_guard = is_<call_guard, T>`, but a proper name is more useful.